pkg/docker/util.go

```diff
@@ -286,6 +286,28 @@ func GetBuilderImage(config *api.Config) (*PullResult, error) {
 	return r, nil
 }
 
+// GetRebuildImage obtains the metadata information for the image
+// specified in a s2i rebuild operation.  Assumptions are made that
+// the build is available locally since it should have been previously built.
+func GetRebuildImage(config *api.Config) (*PullResult, error) {
```

Comment:
Now we have 3 methods that looks very similar, at least, all of them pulling and checking image at the end. I suggest to extract this logic to `pullAndCheck(image string, docker Docker, pullPolicy PullPolicy, forcePull, allowedUIDs boolean) error` method and modify existing methods to invoke it.